.github/workflows/conformance-aws-cni.yaml

```
@@ -115,7 +114,37 @@ jobs:
 
           echo "Generated matrix:"
           cat /tmp/matrix.json
-          echo "matrix=$(jq -c . < /tmp/matrix.json)" >> $GITHUB_OUTPUT
+
+      - name: Install eksctl CLI
+        run: |
+          curl -LO "https://github.com/eksctl-io/eksctl/releases/latest/download/eksctl_$(uname -s)_amd64.tar.gz"
```

Probably better to install the eksctl version already defined in `${{ env.eksctl_version }}` here to avoid surprises when there are breaking changes in future eksctl versions. This way the workflow won't silently start failing on `main` due to an eksctl update but would fail on the respective renovatebot PR bumping the version.

```suggestion
          curl -LO "https://github.com/eksctl-io/eksctl/releases/download/${{ env.eksctl_version }}/eksctl_$(uname -s)_amd64.tar.gz"
```

---

As mentioned in description, supported EKS versions are hardcoded in eksctl and not fetched through some kind of API - that's why I am using newest version here just to determine which EKS versions are available.

---

That's unfortunate.

---

How do we prevent a regression in EKS from breaking our CI?

---

What regression do you have in mind?

---

It already happened several times now that eksctl updates break our CI (one of the reasons for pinning cloud CLIs in our workflows in the first place 😅), so I think this is what André is referring to.

---

I've talked with Andre offline - I've added a comment that we use eksctl without any API interaction - just fetching the list of supported versions directly from binary. 
I've also verified that they are pretty fast with updating supported versions: https://github.com/eksctl-io/eksctl/pull/7781
which was released the same day as 1.30 on EKS.
I've added a link to the upstream issue where they claim to work on API support - once it's implemented, we can switch to that.